pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -207,10 +209,11 @@ autoPatchelf() {
       isELF "$file" || continue
       segmentHeaders="$(LANG=C readelf -l "$file")"
       # Skip if the ELF file doesn't have segment headers (eg. object files).
-      echo "$segmentHeaders" | grep -q '^Program Headers:' || continue
+      # not using grep -q, because it can cause Broken pipe
+      [ -n "$(echo "$segmentHeaders" | grep '^Program Headers:')" ] || continue
       if isExecutable "$file"; then
           # Skip if the executable is statically linked.
-          echo "$segmentHeaders" | grep -q "^ *INTERP\\>" || continue
```

Comment:
This shouldn't ever result in a broken pipe error, at least in this context, so we don't need the workaround here.

Reply:
@danbst: Hm, apparently according to your description it does. Do you have an output of `strace`?

Reply:
@aszlig this is exactly the place where it failed, notice lines:
```
/nix/store/../setup-hook: line 213: echo: write error: Broken pipe
/nix/store/../setup-hook: line 210: echo: write error: Broken pipe
```

Reply:
sorry, missed your last comment. No, I can't reproduce this issue. It probably occurs only on high CPU load/slow CPUs.